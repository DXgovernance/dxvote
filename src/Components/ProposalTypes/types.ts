export interface ProposalTypeDescriptionProps {
  title: string;
  description: string;
  icon?: React.FC<React.SVGProps<SVGSVGElement>>;
  onChainAction: boolean;
}

export interface ProposalTypesProps {
  data: ProposalTypeDescriptionProps[];
}
