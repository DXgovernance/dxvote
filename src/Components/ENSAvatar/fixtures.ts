import { ENSAvatarProps } from './types';

export const withAvatarNFT: ENSAvatarProps = {
  address: '0x0b17cf48420400e1D71F8231d4a8e43B3566BB5B',
};

export const withoutAvatarNFT: ENSAvatarProps = {
  address: '0x519b70055af55A007110B4Ff99b0eA33071c720a',
};

export const customSize: ENSAvatarProps = {
  ...withAvatarNFT,
  size: 100,
};
