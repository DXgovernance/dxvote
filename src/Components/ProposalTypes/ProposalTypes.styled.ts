import styled from 'styled-components';
import { FiX } from 'react-icons/fi';

import { Button } from 'old-components/Guilds/common/Button';
import { Flex } from 'Components/Primitives/Layout';

export const Backdrop = styled(Flex)``;
export const CloseIcon = styled(FiX)`
  color: ${({ theme }) => theme.colors.text};
  height: 1.5rem;
  width: 1.5rem;

  &:hover {
    cursor: pointer;
    opacity: 0.6;
  }
`;

export const Wrapper = styled(Flex)`
  max-width: 386px;
  @media only screen and (max-width: 768px) {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: 1000;
  }
`;
export const Container = styled(Flex)`
  border: 1px solid ${({ theme }) => theme.colors.border.initial};
  margin: 10px 0;
  width: 100%;

  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;

export const PaddingWrapper = styled(Flex)`
  margin: 24px;
  width: 100%;
`;

export const StyledProposalDescription = styled(Flex)`
  display: flex;
  align-items: flex-start;
  margin: 0px 24px;
`;

export const ContentWrapper = styled(Flex)`
  height: 100%;
  justify-content: flex-start;
`;

export const ProposalTypeButton = styled(Button)`
  width: 90%;
  height: 40px;
  margin: 6px 0;
  flex-direction: row;
  justify-content: left;
  &:active,
  &:focus {
    border: 2px solid ${({ theme }) => theme.colors.text};
  }
`;

export const ContainerHeader = styled(Flex)`
  width: 90%;
  justify-content: initial;
  flex-direction: row;
  margin: 6px 0;
  color: ${({ theme }) => theme.colors.proposalText.lightGrey};
`;

export const Footer = styled(Flex)`
  flex-direction: row;
  justify-content: right;
  width: 100%;

  @media only screen and (max-width: 768px) {
    border-top: 1px solid black;
    justify-content: center;
    align-items: center;
    border-radius: 0;
    padding: 20px;
  }
`;

export const ButtonFooter = styled(Button)`
  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;

export const Header = styled(Flex)`
  align-items: flex-start;
  width: 100%;

  @media only screen and (max-width: 768px) {
    border-radius: 0;
    border-bottom: 1px solid black;
    flex-direction: row;
    justify-content: space-between;
    padding: 20px;
    width: 90%;
  }
`;

export const HeaderWrap = styled(Flex)`
  flex-direction: row;
`;

export const TitleWrapper = styled(Flex)`
  width: 100%;
  align-items: flex-start;
  margin-top: 10px;
  color: ${({ theme }) => theme.colors.text};

  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;
