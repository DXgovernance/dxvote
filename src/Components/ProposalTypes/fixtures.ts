import { ProposalTypesProps } from './types';

export const testProps: ProposalTypesProps = {
  data: [
    {
      title: 'Signal Proposal',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed leo quam, blandit eu sapien eu, commodo dapibus nisl',
      onChainAction: false,
    },
    {
      title: 'Custom Proposal',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed leo quam, blandit eu sapien eu, commodo dapibus nisl',
      onChainAction: false,
    },
  ],
};
