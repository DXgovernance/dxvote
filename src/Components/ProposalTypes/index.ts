export { default as ProposalTypes } from './ProposalTypes';
