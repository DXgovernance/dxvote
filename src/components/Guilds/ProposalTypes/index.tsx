import React, { useState } from 'react';
import { AiOutlineArrowLeft } from 'react-icons/ai';
import styled from 'styled-components';
import { FiArrowLeft, FiX } from 'react-icons/fi';
import { Button } from '../common/Button';
import { Flex } from '../common/Layout';
import { ContainerText } from '../common/Layout/Text';
import { useHistory, useLocation } from 'react-router-dom';

import { Heading } from '../common/Typography';
import StyledIcon from '../common/SVG';
import { isMobile } from 'react-device-detect';

const Backdrop = styled(Flex)``;
const CloseIcon = styled(FiX)`
  color: ${({ theme }) => theme.colors.text};
  height: 1.5rem;
  width: 1.5rem;

  &:hover {
    cursor: pointer;
    opacity: 0.6;
  }
`;

const Wrapper = styled(Flex)`
  max-width: 386px;
  @media only screen and (max-width: 768px) {
    position: fixed;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: 1000;
  }
`;
const Container = styled(Flex)`
  border: 1px solid ${({ theme }) => theme.colors.primary};
  margin: 10px 0;
  width: 100%;

  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;

const PaddingWrapper = styled(Flex)`
  margin: 24px;
  width: 100%;
`;

const ContentWrapper = styled(Flex)`
  height: 100%;
  justify-content: flex-start;
`;

const ProposalTypeButton = styled(Button)`
  width: 90%;
  height: 40px;
  margin: 6px 0;
  flex-direction: row;
  justify-content: left;
`;

const ContainerHeader = styled(Flex)`
  width: 90%;
  justify-content: initial;
  flex-direction: row;
  margin: 6px 0;
`;

const Footer = styled(Flex)`
  flex-direction: row;
  justify-content: right;
  width: 100%;

  @media only screen and (max-width: 768px) {
    border-top: 1px solid black;
    justify-content: center;
    align-items: center;
    border-radius: 0;
    padding: 20px;
  }
`;

const ButtonFooter = styled(Button)`
  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;

const Header = styled(Flex)`
  align-items: flex-start;
  width: 100%;

  @media only screen and (max-width: 768px) {
    border-radius: 0;
    border-bottom: 1px solid black;
    flex-direction: row;
    justify-content: space-between;
    padding: 20px;
    width: 90%;
  }
`;

const HeaderWrap = styled(Flex)`
  flex-direction: row;
`;

const TitleWrapper = styled(Flex)`
  width: 100%;
  align-items: flex-start;
  margin-top: 10px;

  @media only screen and (max-width: 768px) {
    width: 90%;
  }
`;

interface ProposalTypeDescriptionProps {
  title: string;
  description: string;
  icon?: React.FC<React.SVGProps<SVGSVGElement>>;
  onChainAction: boolean;
}

const ProposalTypeDescription: React.FC<ProposalTypeDescriptionProps> = ({
  description,
  title,
  onChainAction,
}) => {
  return (
    <Container>
      <PaddingWrapper>
        <ContainerText variant="bold">{title}</ContainerText>
        <ContainerText variant="medium">{description}</ContainerText>
        <ContainerText variant="medium" color="grey">
          {onChainAction ? 'On-chain Action' : 'No on-chain action'}
        </ContainerText>
      </PaddingWrapper>
    </Container>
  );
};

interface ProposalTypesProps {
  data: ProposalTypeDescriptionProps[];
}
const ProposalTypes: React.FC<ProposalTypesProps> = ({ data }) => {
  const history = useHistory();
  const location = useLocation();

  const continueUrl = location.pathname.replace(
    '/proposalType',
    '/create/proposal_type'
  );

  const [proposalDescription, setProposalDescription] =
    useState<ProposalTypeDescriptionProps>(data[0]);

  return (
    <Backdrop>
      <Wrapper>
        {!isMobile && (
          <Header>
            <Button onClick={() => history.push('/')}>
              <StyledIcon margin="0 10px 0 0" src={AiOutlineArrowLeft} />
              Back to overview
            </Button>
          </Header>
        )}
        {isMobile && (
          <>
            <Header onClick={() => history.push('/')}>
              <HeaderWrap>
                <StyledIcon src={FiArrowLeft} />
                Back to overview
              </HeaderWrap>
              <StyledIcon src={CloseIcon} />
            </Header>
          </>
        )}
        <ContentWrapper>
          <TitleWrapper>
            <Heading size={2}>Create Proposal</Heading>
          </TitleWrapper>
          <Container>
            <PaddingWrapper data-testid="proposal-types-list">
              <ContainerHeader>Choose Proposal</ContainerHeader>
              {data.map(({ title, description, onChainAction, icon }) => (
                <ProposalTypeButton
                  onClick={() =>
                    setProposalDescription({
                      title: title,
                      description: description,
                      onChainAction: onChainAction,
                    })
                  }
                >
                  <StyledIcon src={icon} />
                  {title}
                </ProposalTypeButton>
              ))}
            </PaddingWrapper>
          </Container>
          <ProposalTypeDescription
            title={proposalDescription.title}
            description={proposalDescription.description}
            onChainAction={proposalDescription.onChainAction}
            data-testId="proposal-type-description"
          />
        </ContentWrapper>
        <Footer>
          <ButtonFooter
            variant="secondary"
            onClick={() => history.push(continueUrl)}
            data-testid="proposal-type-continue-button"
          >
            Continue
          </ButtonFooter>
        </Footer>
      </Wrapper>
    </Backdrop>
  );
};

export default ProposalTypes;
