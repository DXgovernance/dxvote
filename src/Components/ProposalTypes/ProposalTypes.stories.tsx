import { ComponentStory, ComponentMeta } from '@storybook/react';
import ProposalTypes from 'Components/ProposalTypes/ProposalTypes';
import { testProps } from './fixtures';

export default {
  title: 'Components/ProposalTypes',
  component: ProposalTypes,
  //   argTypes: {},
} as ComponentMeta<typeof ProposalTypes>;

const Template: ComponentStory<typeof ProposalTypes> = args => (
  <ProposalTypes {...args} />
);

export const Default = Template.bind({});
Default.args = testProps;
Default.storyName = 'ProposalTypes';
