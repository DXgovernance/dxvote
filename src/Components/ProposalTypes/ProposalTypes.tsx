import React, { useContext, useState } from 'react';
import { isMobile } from 'react-device-detect';
import { AiOutlineArrowLeft } from 'react-icons/ai';
import { FiArrowLeft } from 'react-icons/fi';
import { useTranslation } from 'react-i18next';

import { Button } from 'old-components/Guilds/common/Button';
import { Loading } from 'Components/Primitives/Loading';
import { useHistory, useLocation } from 'react-router-dom';
import StyledIcon from 'old-components/Guilds/common/SVG';
import { Heading } from 'old-components/Guilds/common/Typography';
import { useTypedParams } from 'Modules/Guilds/Hooks/useTypedParams';
import { GuildAvailabilityContext } from 'contexts/Guilds/guildAvailability';
import { ProposalTypeDescriptionProps, ProposalTypesProps } from './types';
import { ProposalTypeDescription } from './components';
import {
  Container,
  PaddingWrapper,
  Backdrop,
  Wrapper,
  Header,
  HeaderWrap,
  ContentWrapper,
  Footer,
  TitleWrapper,
  ContainerHeader,
  ProposalTypeButton,
  ButtonFooter,
  CloseIcon,
} from './ProposalTypes.styled';

const ProposalTypes: React.FC<ProposalTypesProps> = ({ data }) => {
  const { guildId, chainName: chain } = useTypedParams();
  const { t } = useTranslation();
  const history = useHistory();
  const location = useLocation();
  const { isLoading: isGuildAvailabilityLoading } = useContext(
    GuildAvailabilityContext
  );
  const [proposalDescription, setProposalDescription] =
    useState<ProposalTypeDescriptionProps>(data[0]);

  const continueUrl = location.pathname.replace(
    '/proposalType',
    `/create/${proposalDescription.title}`
  );

  if (isGuildAvailabilityLoading) return <Loading loading />;

  return (
    <Backdrop>
      <Wrapper>
        {isMobile ? (
          <Header onClick={() => history.push(`/${chain}/${guildId}`)}>
            <HeaderWrap>
              <StyledIcon src={FiArrowLeft} />
              {t('backToOverview')}
            </HeaderWrap>
            <StyledIcon src={CloseIcon} />
          </Header>
        ) : (
          <Header>
            <Button
              variant="secondary"
              onClick={() => history.push(`/${chain}/${guildId}`)}
            >
              <StyledIcon margin="0 10px 0 0" src={AiOutlineArrowLeft} />
              {t('backToOverview')}
            </Button>
          </Header>
        )}
        <ContentWrapper>
          <TitleWrapper>
            <Heading size={2}>{t('createProposal')}</Heading>
          </TitleWrapper>
          <Container>
            <PaddingWrapper data-testid="proposal-types-list">
              <ContainerHeader>{t('chooseProposal')}</ContainerHeader>
              {data.map(({ title, description, onChainAction, icon }) => (
                <ProposalTypeButton
                  key={title}
                  variant="secondary"
                  onClick={() =>
                    setProposalDescription({
                      title: title,
                      description: description,
                      onChainAction: onChainAction,
                    })
                  }
                >
                  {icon && <StyledIcon src={icon} />}
                  {title}
                </ProposalTypeButton>
              ))}
            </PaddingWrapper>
          </Container>

          <ProposalTypeDescription
            title={proposalDescription.title}
            description={proposalDescription.description}
            onChainAction={proposalDescription.onChainAction}
            data-testid="proposal-type-description"
          />
        </ContentWrapper>
        <Footer>
          <ButtonFooter
            variant="primary"
            onClick={() => history.push(continueUrl)}
            data-testid="proposal-type-continue-button"
          >
            {t('continue')}
          </ButtonFooter>
        </Footer>
      </Wrapper>
    </Backdrop>
  );
};

export default ProposalTypes;
