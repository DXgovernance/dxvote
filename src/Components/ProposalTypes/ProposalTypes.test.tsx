import ProposalTypes from './ProposalTypes';
import { render } from 'utils/tests';
import { testProps } from './fixtures';

jest.mock('contexts/index', () => jest.fn());

describe('ProposalTypes', () => {
  const props = testProps;

  it('Should match snapshot', () => {
    const { container } = render(<ProposalTypes {...props} />);
    expect(container).toMatchSnapshot();
  });
  it('Should display proposalTypes', () => {
    const { getAllByText } = render(<ProposalTypes {...props} />);
    props.data.forEach(({ title }) => {
      expect(getAllByText(title)[0]).toBeVisible();
    });
  });
});
