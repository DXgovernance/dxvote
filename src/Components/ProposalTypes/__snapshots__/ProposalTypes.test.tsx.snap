// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalTypes Should match snapshot 1`] = `
<div>
   
  <div
    class="sc-pGacB sc-hBEYId imomAO"
  >
    <div
      class="sc-pGacB sc-fFucqa imomAO eKAtoq"
    >
      <div
        class="sc-pGacB sc-iwyWTf imomAO knnsRI"
      >
        <button
          class="sc-bdfBQB cAORUT"
        >
          <div
            class="sc-jSgvzq dMuAyB"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M872 474H286.9l350.2-304c5.6-4.9 2.2-14-5.2-14h-88.5c-3.9 0-7.6 1.4-10.5 3.9L155 487.8a31.96 31.96 0 0 0 0 48.3L535.1 866c1.5 1.3 3.3 2 5.2 2h91.5c7.4 0 10.8-9.2 5.2-14L286.9 550H872c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </div>
          backToOverview
        </button>
      </div>
      <div
        class="sc-pGacB sc-hHfuMS imomAO cEEget"
      >
        <div
          class="sc-pGacB sc-lmoMya imomAO cdlwzX"
        >
          <h2
            class="sc-gKseQn fMfHRo"
            size="2"
          >
            createProposal
          </h2>
        </div>
        <div
          class="sc-pGacB sc-bkzYnD imomAO eCFhqu"
        >
          <div
            class="sc-pGacB sc-idOiZg imomAO Uznir"
            data-testid="proposal-types-list"
          >
            <div
              class="sc-pGacB sc-kfzBvY imomAO cYRbeJ"
            >
              chooseProposal
            </div>
            <button
              class="sc-bdfBQB sc-dmlqKv cAORUT fSVQWC"
            >
              Signal Proposal
            </button>
            <button
              class="sc-bdfBQB sc-dmlqKv cAORUT fSVQWC"
            >
              Custom Proposal
            </button>
          </div>
        </div>
        <div
          class="sc-pGacB sc-bkzYnD imomAO eCFhqu"
        >
          <div
            class="sc-pGacB sc-idOiZg imomAO Uznir"
          >
            <div
              class="sc-pGacB sc-dIUeWJ imomAO hibDnZ"
            >
              <div
                class="sc-pGacB sc-iJuVqt imomAO ggHXoL"
                color="#fff"
              >
                Signal Proposal
              </div>
              <div
                class="sc-pGacB sc-iJuVqt imomAO cfpTdd"
                color="#BDC0C7"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed leo quam, blandit eu sapien eu, commodo dapibus nisl
              </div>
              <div
                class="sc-pGacB sc-iJuVqt imomAO cfpTdd"
                color="#BDC0C7"
              >
                noOnchainAction
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-pGacB sc-fKFxtB imomAO ligvnW"
      >
        <button
          class="sc-bdfBQB sc-bBXrwG cnzpap iljmeA"
          data-testid="proposal-type-continue-button"
        >
          continue
        </button>
      </div>
    </div>
  </div>
   
</div>
`;
